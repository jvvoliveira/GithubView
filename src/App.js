import React, { Component } from 'react'
import './App.css'
import SearchBar from './Components/SearchBar'
import NavBar from './Components/NavBar'
import Router from './Components/Router'
import { BrowserRouter } from 'react-router-dom'
import Loading from './Components/Loading'


export const loadingContext = React.createContext({
  loading: false,
  message: null,
  showLoading: message => { },
  hideLoading: () => { }
})


class App extends Component {

  constructor(props) {
    super(props);

    this.state = {
      repos: [],
      usuario: {},
      loading: false,
      message: null
    }

    this.setRepos = this.setRepos.bind(this);
    this.setUsuario = this.setUsuario.bind(this);
  }

  showLoading = message => {
    this.setState({
      loading: true,
      message
    })
  }

  hideLoading = () => { this.setState({ loading: false }) }

  setRepos(repos) {
    this.setState({ repos });
  }

  setUsuario(usuario) {
    this.setState({ usuario });
  }


  //passando função como prop pro filho...uma maneira de fazer comunicação
  render() {
    const { showLoading, hideLoading } = this

    const value = {
      ...this.state,
      showLoading,
      hideLoading
    }

    return (
      <div className="App">
        <div>
          <BrowserRouter>
            <loadingContext.Provider value={value}>
                <NavBar />
                <SearchBar setRepos={this.setRepos} setUsuario={this.setUsuario}/>
                <Loading/>
                <Router repos={this.state.repos} usuario={this.state.usuario}/>
            </loadingContext.Provider>
          </BrowserRouter>
        </div>
      </div>
    );
  }
}

export default App;
