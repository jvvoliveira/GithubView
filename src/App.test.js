import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import TestRenderer from 'react-test-renderer';
import renderer from 'react-test-renderer';

describe("initial test App", () =>{
  it('should renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('should set state repos value', () => {
    const repos = [
      {
        name:"repo1",
        description:"primeiro repositório de teste",
        link:"www.google.com",
        language:"java"
      }
    ];
    const testRenderer = TestRenderer.create(
      <App></App>
    );

    let app = testRenderer.getInstance();
    app.setRepos(repos);

    expect(app.state.repos[0].name).toEqual("repo1");
    expect(app.state.repos[0].description).toEqual("primeiro repositório de teste");
    expect(app.state.repos[0].link).toEqual("www.google.com");
    expect(app.state.repos[0].language).toEqual("java");

  });
  it("snapshot",()=>{
    const tree = renderer.create(<App></App>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

