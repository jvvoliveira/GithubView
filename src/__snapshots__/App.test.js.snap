// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial test App snapshot 1`] = `
<div
  className="App"
>
  <div
    className="search"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder="nome do usuÃ¡rio no github"
        type="text"
      />
      <input
        onClick={[Function]}
        type="submit"
        value="pesquisar"
      />
    </form>
  </div>
</div>
`;
